ServerCore/SyncController.cs

```diff
@@ -489,8 +484,74 @@ public SyncController(PuzzleServerContext context, UserManager<IdentityUser> use
 
             var helper = new SyncHelper(context);
             var response = await helper.GetSyncResponse(currentEvent.ID, team.ID, puzzleId, query_puzzle_ids, min_solve_count,
-                                                        annotations, last_sync_time);
+                                                        annotations, last_sync_time, false);
             return Json(response);
         }
+        
+        // GET api/Sync/client
+        [HttpGet]
+        [Route("{eventId}/{puzzleId}/api/Sync/client")]
+        public async Task<IActionResult> Index(string eventId, int puzzleId)
+        {
+            Event currentEvent = await EventHelper.GetEventFromEventId(context, eventId);
+            if (currentEvent == null)
+            {
+                return Content("ERROR:  That event doesn't exist");
```

Comment:
You can return NotFound(), Forbid(), and things like this so you get proper return codes. Those take strings as well so you can add the ones you have.

Reply:
Ah, thanks for the info.  I was annoyed by the many pages on the website that just give an ACCESS DENIED or FORBIDDEN error without giving the slightest clue what the problem is, which is why I did it this way.  But it's nice to know there's a way to get the best of both worlds, a nicely-formatted error that still has some useful information.

Reply:
Well, I tried using Forbid("..."), but it produced really ugly error messages.  An example of what the user sees is below.  So I'll stick with the informative error messages I have.

InvalidOperationException: No authentication handler is registered for the scheme 'ERROR: That event doesn't exist'. The registered schemes are: Identity.Application, Identity.External, Identity.TwoFactorRememberMe, Identity.TwoFactorUserId, Microsoft. Did you forget to call AddAuthentication().Add[SomeAuthHandler]("ERROR: That event doesn't exist",...)?